Student/StudentUnitTests/Assignments/AssignmentList/AssignmentListViewControllerTests.swift

```diff
@@ -0,0 +1,287 @@
+        XCTAssertEqual(cell?.textLabel?.text, "how to test")
+
+        XCTAssertEqual( vc.gradingPeriodLabel.text, "grading period z" )
+        XCTAssertEqual( vc.filterButton.title(for: .normal), "Clear filter" )
+    }
+
+    func testSelectFirstOnIpad() {
+        let svc = UISplitViewController(nibName: nil, bundle: nil)
+        let nav = UINavigationController(rootViewController: vc)
+        svc.viewControllers = [nav, UIViewController()]
+
+        gradingPeriods = [ APIAssignmentListGradingPeriod.make(title: "grading period a") ]
+
+        let assignmentsGroupA = [
+            APIAssignmentListAssignment.make(id: "1", name: "ios 101", dueAt: Date().addDays(1)),
+        ]
+
+        let assignmentsGroupB = [
+            APIAssignmentListAssignment.make(id: "2", name: "how to cook pizza"),
+        ]
+
+        groups = [
+            APIAssignmentListGroup.make(id: "1", name: "GroupA", assignments: assignmentsGroupA),
+            APIAssignmentListGroup.make(id: "2", name: "GroupB", assignments: assignmentsGroupB),
+        ]
+
+        req = AssignmentListRequestable(courseID: courseID, gradingPeriodID: nil, filter: false)
+        mockNetwork()
+
+        loadView()
+
+        let expected = IndexPath(row: 0, section: 0)
+        let cell = vc.tableView.cellForRow(at: expected) as? AssignmentListViewController.ListCell
+        XCTAssertTrue(cell?.isSelected ?? false)
+        XCTAssertEqual(expected, vc.tableView.indexPathForSelectedRow)
+    }
+
+    func testPaging() {
+        gradingPeriods = [ APIAssignmentListGradingPeriod.make(title: "grading period a") ]
+        var assignmentsGroupA1 = [APIAssignmentListAssignment]()
+        var assignmentsGroupA2 = [APIAssignmentListAssignment]()
+
+        var count = 5
+        for i in 0..<count {
+            let a = APIAssignmentListAssignment.make(id: ID("\(i)"), name: "\(i)")
+            assignmentsGroupA1.append(a)
+        }
+
+        count = 6
+        for i in 0..<count {
+            let a = APIAssignmentListAssignment.make(id: ID("10\(i)"), name: "\(5+i)")
+            assignmentsGroupA2.append(a)
+        }
+
+        groups = [
+            APIAssignmentListGroup.make(id: "1", name: "GroupA", assignments: assignmentsGroupA1, pageInfo: APIPageInfo(endCursor: "MQ", hasNextPage: true)),
+        ]
+
+        req = AssignmentListRequestable(courseID: courseID, gradingPeriodID: nil, filter: false)
+
+        let d1 = data(gradingPeriods: gradingPeriods, groups: groups)
+
+        groups = [
+            APIAssignmentListGroup.make(id: "1", name: "GroupA", assignments: assignmentsGroupA2),
+        ]
+        let d2 = data(gradingPeriods: gradingPeriods, groups: groups)
+        var cnt = 0
+        api.mock(req, data: nil, response: nil, error: nil, dataHandler: { () -> MockURLSession.UrlResponseTuple in
+            if cnt == 0 {
+                cnt += 1
+                return (d1, nil, nil)
+            } else {
+                return (d2, nil, nil)
+            }
```

Comment:
Did you need this because you couldn't prevent the view from automatically getting the next page?

Reply:
I needed to support testing paging.  On our current MockSession only supports 1 payload for the same url.  Graphql api's all have the same url